src/test/java/com/videoclub/suriken/api/MovieControllerTest.java

```diff
@@ -0,0 +1,156 @@
+package com.videoclub.suriken.api;
+
+import com.videoclub.suriken.model.Genre;
+import com.videoclub.suriken.model.Movie;
+import com.videoclub.suriken.model.MovieRenter;
+import com.videoclub.suriken.repository.MovieRepository;
+import com.videoclub.suriken.repository.RenterRepository;
+import org.junit.jupiter.api.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.web.client.TestRestTemplate;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import static org.assertj.core.api.AssertionsForClassTypes.assertThat;
+
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+class MovieControllerTest {
+
+    Logger logger = LoggerFactory.getLogger(MovieControllerTest.class);
+
+    @Autowired
+    private MovieRepository movieRepository;
+
+    @Autowired
+    private RenterRepository renterRepository;
+
+    @Autowired
+    private TestRestTemplate restTemplate;
+
+//    @BeforeAll
+//    static void beforeAll() {
+//        Movie movie1 = new Movie(null, "Titanik", 1995, Genre.DRAMA, "James Camoron", 3);
+//        Movie movie2 = new Movie(null, "The Shawshank Redemption", 1994, Genre.DRAMA, "Frank Darabont", 5);
+//        Movie movie3 = new Movie(null, "The Godfather", 1972, Genre.CRIME, "Francis Ford Coppola", 1);
+//        Movie movie4 = new Movie(null, "The Dark Knight", 2008, Genre.ACTION, "Christopher Nolan", 7);
+//    }
+
+    @Test
+    void successfullyAddMovie() {
+        HttpEntity<Movie> request = new HttpEntity<>(new Movie(null, "Titanik", 1995, Genre.DRAMA, "James Camoron", 3));
+        ResponseEntity response = restTemplate.postForEntity("/api/v1/movie", request, Void.class);
+
+        assertThat(response.getStatusCode()).isEqualTo(HttpStatus.CREATED);
+
+        Optional<Movie> respMovie = movieRepository.findMovieByName("Titanik");
+//        logger.info("Id of saved movie is {}", respMovie.get().getId());
+        assertThat(respMovie.isPresent()).isEqualTo(true);
+    }
+
+    @Test
+    void getAMovieWhenExist() {
+        movieRepository.save(new Movie(null, "Titanik", 1995, Genre.DRAMA, "James Camoron", 3));
+
+        ResponseEntity<Movie> response = restTemplate.getForEntity("/api/v1/movie/1", Movie.class);
+
+        assertThat(response.getStatusCode()).isEqualTo(HttpStatus.OK);
```

Comment:
Proverio bih ovde da li je to sto je vraceno ono sto ocekujes da se vrati, tj ovo sto si na liniji 62 i sacuvao